File: tests_oracle.ts
import { OracleClient } from '../src/OracleClient';
import { ApplicationClient } from '@algorandfoundation/algokit-utils';
import { Algodv2 } from 'algosdk';

// Mock Algod client
const algod = new Algodv2('token', 'http://localhost', 4001);

// Mock ApplicationClient configuration
const appClient = new ApplicationClient({
  app: {
    name: 'OracleApp',
    methods: [
      { name: 'update', args: ['uint64', 'byte[]'] },
      { name: 'get', args: ['uint64'] }
    ]
  },
  algod: algod,
  sender: {
    addr: 'TESTACCOUNT',
    signer: () => Promise.resolve(new Uint8Array())
  }
});

test("Oracle workflow", async () => {
    const oracle = new OracleClient(appClient);
    
    // Test data update
    await oracle.updateData(100, new TextEncoder().encode("price=1.2"));
    
    // Test valid retrieval
    const data = await oracle.getData(100);
    expect(new TextDecoder().decode(data)).toBe("price=1.2");
    
    // Test missing data error
    await expect(oracle.getData(101)).rejects.toThrow();
});

File: tests_oracle.py
def test_oracle_flow():
    # Deploy contract
    client = OracleApp(algod_client)
    oracle_client = OracleClient(client)

    # Update data
    oracle_client.update_data(round=100, data=b"price=1.2")

    # Test get
    assert oracle_client.get_data(100) == b"price=1.2"
    assert oracle_client.get_data(101) == b""

    # Test must_get (expect panic)
    with pytest.raises(Exception):
        oracle_client.must_get_data(101)